src/app/features/asset-list/stacks/stx20-token-asset-list/stx20-token-asset-list.tsx

```diff
@@ -10,8 +12,16 @@ interface Stx20TokenAssetDetails {
 
 interface Stx20TokenAssetListProps {
   tokens: Stx20TokenAssetDetails[];
+  countSetter: React.Dispatch<React.SetStateAction<number>>;
 }
-export function Stx20TokenAssetList({ tokens }: Stx20TokenAssetListProps) {
+export function Stx20TokenAssetList({ tokens, countSetter }: Stx20TokenAssetListProps) {
+  useEffect(() => {
+    if (countSetter) {
+      countSetter(tokens.length);
+    }
+  }, [tokens, countSetter]);
```

Comment:
Why do we need this `countSetter`? Can't we infer the value another way to avoid having to do this?

The parent passes in `tokens`, so it knows the `length` already.

Reply:
Handled them in the bitcoin & stacks list components. happy to take feedback on a cleaner way to have the manage-tokens component know if the children have any tokens.

```
{
                useEffect(() => {
                  if (hasTokenSetter && tokens.length) hasTokenSetter(true);
                });
                return <Stx20TokenAssetList tokens={tokens} />;
              }}
```

Reply:
above comment is no longer up-to-date. I had to re-add them to the [token]-asset-lists components above.

```
  useEffect(() => {
    if (hasTokenSetter && tokens.length) hasTokenSetter(true);
  }, [tokens.length, hasTokenSetter]);
```

The state update needs to happen in a use-effect. hence cant be done in a higher component.
```
sip10-token-asset-list.tsx:28 Warning: Cannot update a component (`ManageTokensDialog`) while rendering a different component (`Sip10TokenAssetList`). To locate the bad setState() call inside `Sip10TokenAssetList`, follow the stack trace as described in 
```
https://stackoverflow.com/questions/62336340/cannot-update-a-component-while-rendering-a-different-component-warning

happy to get feedback here on a cleaner approach. we can potentially use redux instead of lifting the state up.

@kyranjamie @pete-watters